metrics-tracing-context/src/tracing_integration.rs

```diff
@@ -123,8 +131,7 @@ where
             .expect("subscriber should downcast to expected type; this is a bug!");
         let span = subscriber.span(id).expect("registry should have a span for the current ID");
 
-        let result =
-            if let Some(labels) = span.extensions().get::<Labels>() { f(labels) } else { None };
+        let result = span.extensions().get::<Labels>().and_then(f);
         result
```

Comment:
We should just do:

```suggestion
span.extensions().get::<Labels>().and_then(f)
```

I'm not sure why I/whoever originally split it across two lines. 🤔

Reply:
How did I miss it :smile:

Reply:
Oh, I know why:

```rust
error[E0597]: `span` does not live long enough
   --> metrics-tracing-context/src/tracing_integration.rs:134:9
    |
132 |         let span = subscriber.span(id).expect("registry should have a span for the current ID");
    |             ---- binding `span` declared here
133 |
134 |         span.extensions().get::<Labels>().and_then(f)
    |         ^^^^-------------
    |         |
    |         borrowed value does not live long enough
    |         a temporary with access to the borrow is created here ...
135 |     }
    |     -
    |     |
    |     `span` dropped here while still borrowed
    |     ... and the borrow might be used here, when that temporary is dropped and runs the destructor for type `Extensions<'_>`
    |
    = note: the temporary is part of an expression at the end of a block;
            consider forcing this temporary to be dropped sooner, before the block's local variables are dropped
help: for example, you could save the expression's value in a new local variable `x` and then make `x` be the expression at the end of the block
    |
134 |         let x = span.extensions().get::<Labels>().and_then(f); x
    |         +++++++                                              +++

```

That's weird that by just assigning it to a binding the error is resolved.

So either leave as is, or:

```rust
        let ext = span.extensions();
        ext.get::<Labels>().and_then(f)
```

I would go with the latter as it is more ... easier to get? And is a common pattern, because `.extensions()` is the source of the temporary in the chained methods expression.

Reply:
Oh, well that's annoying. Yeah, I agree with your idea; let's go with that approach.